library/core/src/alloc/mod.rs

```diff
@@ -50,13 +42,13 @@ pub struct MemoryBlock {
 ///
 /// ### Currently allocated memory
 ///
-/// Some of the methods require that a memory block be *currently allocated* via an allocator. This
+/// Some of the methods require that a memory slice be *currently allocated* via an allocator. This
```

Comment:
I don't think we should change the documentation to "slice" in this file. "Memory block" is still a valid term even if we don't use an explicit `MemoryBlock` type. Using "slice" might be confusing since most people think of slices as borrowed rather than owned.

Reply:
> Using "slice" might be confusing since most people think of slices as borrowed rather than owned.

Fair point, reverted it.